agent/consul/leader_connect_ca.go

```diff
@@ -1498,11 +1498,6 @@ func (c *CAManager) SignCertificate(csr *x509.CertificateRequest, spiffeID conne
 		pem = pem + ca.EnsureTrailingNewline(p)
 	}
 
-	// Append our local CA's intermediate if there is one.
-	if inter != root {
-		pem = pem + ca.EnsureTrailingNewline(inter)
-	}
```

Comment:
Calling out this change loudly. I'm pretty sure this is safe, but I would appreciate some close attention to be sure I'm not missing something. Echo'ing the commit message here:

> Immediately above this line we are already appending the full list of
intermediates. The `provider.ActiveIntermediate` MUST be in this list of
intermediates because it must be available to all the other non-leader
Servers.  If it was not in this list of intermediates then any proxy
that received data from a non-leader would have the wrong certs.

> This is being removed now because we are planning on changing the
`Provider.ActiveIntermediate` interface, and removing these extra calls ahead of
time helps make that change easier.

From what I could tell in git history this logic was part of the initial implementation, so likely just an oversight that we haven't noticed yet.

Reply:
Hmm, because of the bug described in #10532, this change is not safe. We'd have to fix that first for this change to be ok.

Reply:
#11671 fixes the problem that this is masking, so rebasing this PR on that change makes this safe once again.